keps/0015-dry-run.md

```diff
@@ -53,10 +53,52 @@ either as an argument for built-in admission plugins or through a dryRun
 query-parameter for dynamic webhooks, will give them a chance not to have any
```

Comment:
clarify the following:

> While admission controllers are not supposed to have side-effects when triggered, some of them still do. Quotas for example uses the side-effects when triggered

Side-effects are not prohibited as long as there is a reconciliation loop that accounts for the admission plugin getting called and the mutation not being persisted. The reason we want to provide a dry run indicator to admission plugins is so they can improve user experience in cases where waiting for that reconciliation would be disruptive (like quota).